### DISCUSSION TITLE
Autofocus en boton dentro de un Modal

### DISCUSSION THREAD

**Discussion by nachozullo** [TiendaNube/nimbus-design-system#214] [Q&A] [answered]
Buenas, detectamos que estamos teniendo un autofocus en el botón de cancelar dentro de los Modals de nimbus 2. De nuestra parte analizamos y no hay nada que pueda estar generando ese focus. Parece ser algo propio de la implementación del Modal, lo podríamos revisar? 

Les dejo algunos ejemplos: 

![image](https://github.com/TiendaNube/nimbus-design-system/assets/28366930/6c4681c0-57d9-453b-bcfa-43f54f4d9771)

![image](https://github.com/TiendaNube/nimbus-design-system/assets/28366930/e9e69be0-f30f-403f-b184-5bfa1c270603)

![image](https://github.com/TiendaNube/nimbus-design-system/assets/28366930/15ee8866-f12b-430e-9649-c06030b6fc70)

Vimos que ya en storybook se replica este comportamiento:  https://tiendanube.github.io/nimbus-design-system/?path=/story/composite-modal--basic&args=open:!true

![image](https://github.com/TiendaNube/nimbus-design-system/assets/28366930/104e98f8-901e-4839-9a2d-68cbd2a41c29)

**Comment by juanchigallego** [comment]
Hola @nachozullo , cómo estás?
Este comportamiento se da porque al renderizar en un portal el foco del Modal toma prioridad sobre el contenido principal del DOM, y el primer elemento con posibilidad de focus es el botón cancelar que figura primero.
Para poder alterar este comportamiento se puede utilizar la propiedad `tabindex` [nativa de los elementos HTML](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/tabindex) para establecer el orden de foco de los elementos que deseen.
Como por defecto el modal es un componente de contenedor, toda esta lógica queda del lado de la implementación para que puedan lograr la experiencia que busquen.

Avisame si con esta info pudieron modificar el foco del botón cancelar. Saludos!

**Reply by nachozullo** [reply]
Hola @juanchigallego! Gracias por la respuesta! 

Y no tenemos alguna forma de no focusear ningun elemento? Si le pongo tabIndex="-1" a los botones de confirmar/cancelar me termina focuseando la cruz de cerrar el modal. 

Tenes idea si se podría resolver? 

Si la idea es que siempre al desplegarse el modal tenga algun elemento en focus, me lo llevo para hablarlo con charly y barbi

**Reply by juanchigallego** [reply, marked as answer]
Por un tema de accesibilidad es bueno que todos los elementos se puedan focusear para permitirle al usuario navegar con el teclado, en el caso de no poder utilizar un mouse. Con lo cual, lo ideal es no retirarle el foco a los botones. Tampoco podemos quitar el foco del botón de cerrar porque de esta forma estaríamos impidiendo que un usuario que usa teclado pueda cerrar el modal.
Lo que te sugeriría en tal caso es determinar un orden de tabindex donde la primera acción en quedar en foco sea la primaria y no la secundaria.

Te dejo un ejemplo en un [sandbox](https://codesandbox.io/p/sandbox/modal-tabindex-lqffvj?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clqfevbmh00063b6e4ghi6nti%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clqfevbmh00023b6ea6vgl03y%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clqfevbmh00033b6edqwmctzc%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clqfevbmh00053b6eaxbqrtie%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clqfevbmh00023b6ea6vgl03y%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clqfevbmh00013b6eaub2u5dd%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fpublic%252Findex.html%2522%257D%255D%252C%2522id%2522%253A%2522clqfevbmh00023b6ea6vgl03y%2522%252C%2522activeTabId%2522%253A%2522clqfevbmh00013b6eaub2u5dd%2522%257D%252C%2522clqfevbmh00053b6eaxbqrtie%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clqfevbmh00043b6e87t84i3d%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clqfevbmh00053b6eaxbqrtie%2522%252C%2522activeTabId%2522%253A%2522clqfevbmh00043b6e87t84i3d%2522%257D%252C%2522clqfevbmh00033b6edqwmctzc%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clqfevbmh00033b6edqwmctzc%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D).

Cualquier otra duda, avisame!
